mkosi/backend.py

```diff
@@ -619,6 +621,48 @@ def get_partition(self, ident: PartitionIdentifier) -> Optional[Partition]:
             return None
         return self.partition_table.partitions.get(ident)
 
+    def _output_split_path(self, path_pattern: Optional[Path], partition_identifier: PartitionIdentifier) -> Path:
+        assert path_pattern is not None
```

Comment:
I haven't followed the logic, but if this is not supposed to be called with `Non` as `path_pattern`, then let's just exclude it in the signature, so this is a type checking time error instead of a runtime exception.
```suggestion
    def _output_split_path(self, path_pattern: Path, partition_identifier: PartitionIdentifier) -> Path:
```